src/ui/qgsdatasourcemanagerdialog.ui

```diff
@@ -131,12 +125,18 @@
        </size>
       </property>
       <property name="frameShape">
-       <enum>QFrame::StyledPanel</enum>
+       <enum>QFrame::NoFrame</enum>
       </property>
       <property name="frameShadow">
        <enum>QFrame::Raised</enum>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_2" stretch="0">
+       <property name="spacing">
+        <number>6</number>
+       </property>
+       <property name="margin">
+        <number>0</number>
```

Comment:
This is really fragile, because Qt designer likes to forget this setting and reset it back to the default. You're better off setting it in qgsdatasourcemanagerdialog.cpp, line 37 after the call to setupUi.

    ui->verticalLayout_2->setSpacing(6);
    ui->verticalLayout_2->setMargin(0);
    ui->verticalLayout_2->setContentsMargins(0,0,0,0);